### DISCUSSION TITLE
Table Objects temporary not with prim. key auto. Increment

### DISCUSSION THREAD

**Discussion by KristofKlein** [StefanMaron/BusinessCentral.LinterCop#22] [New Rule]
A table object can be set - per definition - to be temporary only.

It behaves as a normal table object while it gets designed. But not while runtime! We found out, that it allows you to define a primary key with auto increment. But since the auto increment is a feature that the SQL server executes it is actually not working for temp tables (like in CAL). 
So It would be nice to get a hint, that it is not possible to use an auto. inc. ...or guide into the direction to use a number sequence instead?

**Comment by StefanMaron** [comment]
This would even make sense to raise an error instead of a warning
